Neuron/Services/Common/TxStatusManager.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  TransactionStatusManager.swift
+//  Neuron
+//
+//  Created by 晨风 on 2018/11/16.
+//  Copyright © 2018 Cryptape. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+
+enum TransactionStateResult {
+    case pending
+    case success(transaction: TransactionDetails)
+    case failure
+}
+
+class TxStatusManager: NSObject {
+    static let manager = TxStatusManager()
+    static let didUpdateTxStatus = Notification.Name("didUpdateTxStatus")
+    static let didAddLocationTxDetails = Notification.Name("didAddLocationTxDetails")
+    static let transactionKey = "transaction"
+    private var localTxDetailList = [LocalTxDetailModel]()
+    private var taskThread = TaskThread()
+    private var realm: Realm!
+
+    private override init() {
+        super.init()
+        configureTxStatusManager()
+        checkSentTransactionStatus()
+    }
+
+    private func configureTxStatusManager() {
+        guard taskThread.thread == nil else { return }
+        taskThread.run()
+        taskThread.syncPerform {
+            self.realm = try! Realm()
+            self.localTxDetailList = self.realm.objects(LocalTxDetailModel.self).filter({ (transaction) -> Bool in
+                return transaction.status == .pending
+            })
+        }
+    }
+
+    // MARK: - Add transaction
+    func insertLocalTxDetail(cetateModelBlock: @escaping () -> LocalTxDetailModel) {
+        configureTxStatusManager()
+        taskThread.perform {
+            let localTxDetail = cetateModelBlock()
+            try? self.realm.write {
+                self.realm.add(localTxDetail)
+            }
+            self.localTxDetailList.append(localTxDetail)
+            NotificationCenter.default.post(name: TxStatusManager.didAddLocationTxDetails, object: nil, userInfo: [TxStatusManager.transactionKey: localTxDetail.getTransactionDetails()])
```

Comment:
I suggest we send out notification in main thread that observers won't need to care about if they're coming from background thread.